src/main/java/org/wso2/securevault/secret/SecretManager.java

```diff
@@ -214,6 +194,73 @@ public void init(Properties properties) {
         initialized = true;
     }
 
+    /**
+     * Split the secret annotation from the delimiter provided and decides whether to use the legacy provider
+     * or a novel provider to obtain the secret value of the requested alias.
+     *
+     * @param secretAnnotation String contains the alias, the provider type and the repository type.
+     * @return Plain text value of the required secret.
+     */
+    public String resolveSecret(String secretAnnotation) {
+
+        String[] secretAnnotationStrings = secretAnnotation.split(DELIMITER);
+        if (secretAnnotationStrings.length == 1) {
+            if (isLegacyProvidersExists) {
```

Comment:
isLegacyProvidersExists doesn't sound correct.

Reply:
Changed to legacyProvidersExist